### ISSUE TITLE
[Bug]: skipped darwin/arm64 build on Go 1.17

### ISSUE THREAD

**Issue by tczekajlo** [goreleaser/goreleaser#2412]
### What happened?

A build for darwin/arm64 is skipped if Go 1.17 is used.

### How can we reproduce this?

Run a build for darwin/arm64 with Go 1.17

### goreleaser version

```bash
goreleaser version 0.174.2
commit: 5227ef0c245de76da6e80e0111ed303252731b8c
built at: 2021-08-02T00:07:00Z
built by: goreleaser
module version: v0.174.2, checksum: h1:EltDZkVGq0mC2Aqgc7uGT7A0aaXdt4KPV6OxTmUG4pk=

https://goreleaser.com
```

### GoReleaser Check

- [X] goreleaser check shows no errors

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

```
   • loading config file       file=.goreleaser.yml
   • checking config:
      • snapshotting
      • github/gitlab/gitea releases
      • project name
      • loading go mod information
      • building binaries
         • DEPRECATED: skipped darwin/arm64 build on Go < 1.16 for compatibility, check https://goreleaser.com/deprecations/#builds-for-darwinarm64 for more info.
      • creating source archive
      • archives
      • linux packages
      • snapcraft packages
      • calculating checksums
      • signing artifacts
      • docker images
      • docker manifests
      • artifactory
      • blobs
      • homebrew tap formula
      • scoop manifests
      • twitter
      • milestones
   • config is valid
```

**Comment by caarlos0** [comment]
It's already fixed on master, will be out in the next release

**Comment by caarlos0** [comment]
should be fixed in v0.175+
